File: quill_native_bridge/quill_native_bridge_android/android/src/main/kotlin/dev/flutterquill/quill_native_bridge/QuillNativeBridgePlugin.kt
package dev.flutterquill.quill_native_bridge

import dev.flutterquill.quill_native_bridge.generated.QuillNativeBridgeApi
import io.flutter.embedding.engine.plugins.FlutterPlugin

class QuillNativeBridgePlugin : FlutterPlugin {
    private var api: QuillNativeBridgeApi? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        api = QuillNativeBridgeImpl(binding.applicationContext)
        requireNotNull(api) { "A new instance of $QuillNativeBridgeApi was created that appeared to be null" }
        QuillNativeBridgeApi.setUp(binding.binaryMessenger, api)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        QuillNativeBridgeApi.setUp(binding.binaryMessenger, api)
        api = null
    }
}

File: quill_native_bridge/quill_native_bridge_ios/ios/Classes/QuillNativeBridgePlugin.swift
import Flutter
import UIKit

public class QuillNativeBridgePlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let messenger = registrar.messenger()
        let api = QuillNativeBridgeImpl()
        QuillNativeBridgeApiSetup.setUp(binaryMessenger: messenger, api: api)
    }
}
